File: src/transformers/features/notations/NotationCreateAssertGeneralTransformer.ts
import { NotationAssertGeneralProgrammer } from "../../../programmers/notations/NotationAssertGeneralProgrammer";

import { StringUtil } from "../../../utils/StringUtil";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace NotationCreateAssertGeneralTransformer {
  export const transform =
    (rename: (str: string) => string) => (props: ITransformProps) =>
      GenericTransformer.factory({
        ...props,
        method: `notations.createAssert${StringUtil.capitalize(rename.name)}`,
        write: (x) =>
          NotationAssertGeneralProgrammer.write({
            ...x,
            rename,
          }),
      });
}

File: packages/transform/src/features/notations/NotationCreateAssertGeneralTransformer.go
package notations

import (
	corenotations "github.com/samchon/typia/packages/core/src/programmers/notations"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var NotationCreateAssertGeneralTransformer = notationCreateAssertGeneralTransformerNamespace{}

type notationCreateAssertGeneralTransformerNamespace struct{}

func (notationCreateAssertGeneralTransformerNamespace) Transform(rename string) internal.Task {
	return func(call internal.ITypiaCallExpression) string {
		return corenotations.NotationAssertGeneralProgrammer.Write(rename, call.TypeText)
	}
}
